### PR TITLE
NEXT-9292 Fix bug for product image positions to normalize sorting an…

### PR DESCRIPTION
**Pull request by JoshuaBehrens**
### 1. Why is this change necessary?
When you add a product with multiple images via sidebar the images that are not the first don't get a position assigned. When no position is assigned along the way for the data to the database only the database default value acts here and sets it to 1. When you now have duplicate positions sortings and galleries behave unexpected.

### 2. What does this change do, exactly?
It uses now the same procedure to set the position for a product_media entity in the administration.
At one point I noticed the cover is set twice and I leveled this as well.

### 3. Describe each step to reproduce the issue or behaviour.
1. Create/edit product
2. Add at least three images via sidebar
3. Save
4. Look into database or in the storefront

### 4. Please link to the relevant issues.
NEXT-9292

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by jleifeld** [comment]
Your change looks good 👍  Can you write a unit tests for this?

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+title:         Fix bug when images are add to a product the position 1 can be assigned multiple times
+issue:         NEXT-9292
+author:        Joshua Behrens
+author_email:  code@joshua-behrens.de
+author_github: JoshuaBehrens
+___
+# Administration
+* Match procedure to assign `product_media.position` in `sw-product-media-form` and `sw-product-detail` so it is always set to prevent a bug where the position 1 is automatically assigned which results in unexpected sorting and selection behaviours
+___
+# Upgrade Information
```
Comment (mitelg) [inline changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:11]: Upgrade information is meant for third party developers, if they have to change their code according to the new changes. 

I don't know if this really matches here. @Phil23 ?

Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:11]: Oh I thought for any shop owners. Ambiguity 🎉

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---
+title:         Fix bug when images are add to a product the position 1 can be assigned multiple times
+issue:         NEXT-9292
+author:        Joshua Behrens
+author_email:  code@joshua-behrens.de
+author_github: JoshuaBehrens
+___
+# Administration
+* Match procedure to assign `product_media.position` in `sw-product-media-form` and `sw-product-detail` so it is always set to prevent a bug where the position 1 is automatically assigned which results in unexpected sorting and selection behaviours
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:9]: Please use `Changed`, `Added`, `Removed` or `Deprecated` here. :-)

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md (hunk 1 above)
Comment (Phil23) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:11]: Also the headline should be more topic related, because nobody will know, what "this bug" actually is.

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+---
+title:         Fix bug when images are add to a product the position 1 can be assigned multiple times
+issue:         NEXT-9292
+author:        Joshua Behrens
+author_email:  code@joshua-behrens.de
+author_github: JoshuaBehrens
+___
```
Comment (Phil23) [inline changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:7]: Also the template is not right. The meta information have to be wrapped by `---` and not `___`.

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md (hunk 2 above)
Comment (Phil23) [inline changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:9]: Please make sure to use one of the defined keywords. 

https://github.com/shopwareBoostDay/platform/blob/master/changelog/readme.md

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md (hunk 1 above)
Comment (Phil23) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:11]: And the "Upgrade Information" should only describe, what developers have to change when upgrading to the specific version.

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md (hunk 3 above)
Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:7]: I copied it strait from the _example.md

grafik

It contains the underscores as well.

Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:7]: Shall I fix it as well? :)

Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:7]: Nvm I deleted the wrong separator 🥴

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md (hunk 1 above)
Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:11]: Is there any other place where I should place the SQL? I mean there were upgrade guides for agencies/shop owners in an update for shopware 5. I'd like to keep it there so people are aware of this.

Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:11]: You mean the headline is not topic specific when it is merged into the compiled markdown?

File: changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title:         Fix bug when images are add to a product the position 1 can be assigned multiple times
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:2]:
```suggestion
title:         Fix bug when images are added to a product the position 1 can be assigned multiple times
```

Comment (JoshuaBehrens) [reply changelog/_unreleased/2020-09-17-fix-product-media-position-duplication.md:2]: Good catch

**Comment by JoshuaBehrens** [comment]
@jleifeld I am unsure how to test it. I see a good test that I could extend but I don't see how I can read vue variables to validate the position data: src/Administration/Resources/app/administration/test/e2e/cypress/integration/media-marketing/sw-product/edit-media.spec.js

**Comment by shyim** [comment]
As we need test here and none are still present. I will close this then here.

@jleifeld Can you maybe answer the question, so he could create an updated one at shopware/platform? :pray:
